code/firmware/rosco_m68k_firmware/bootstrap.asm

```diff
@@ -325,23 +334,28 @@ BERR_HANDLER::
     rte
 
 
-; Convenience to install temporary BERR handler
+; Install temporary BERR handler
 ; Zeroes bus error flag (at BERR_FLAG) and stores old handler
 ; for a subsequent RESTORE_BERR_HANDLER.
 INSTALL_TEMP_BERR_HANDLER::
     move.b  #0,BERR_FLAG                ; Zero bus error flag
     move.l  $8,BERR_SAVED               ; Save the original bus error handler
-    move.l  #BERR_HANDLER,$8            ; Install temporary bus error handler
+    cmpi.b  #$20,SDB_CPUINFO+0          ; Compare the CPU model in SDB (highest 3 bits) against 1
+    blt     .IS000                      ; Is it an MC68000?
+.NOT000
+    move.l  #BERR_HANDLER_MC680X0,$8    ; Install other MC680x0 temporary bus error handler
+    rts
+.IS000
+    move.l  #BERR_HANDLER_MC68000,$8    ; Install MC68000 temporary bus error handler
     rts
 
 
-; Convenience to restore BERR handler from C, after a
-; call to INSTALL_TEM_BERR_HANDLER.
+; Restore BERR handler, after a call to INSTALL_TEM_BERR_HANDLER.

```

Comment:
nitpick: typo (`TEM` vs `TEMP`)

Reply:
Thanks for having a look! Good catch on the typo, I've fixed that.